drivers/frequency/hmc7044/hmc7044.c

```diff
@@ -357,28 +357,36 @@ int32_t hmc7044_clk_round_rate(struct hmc7044_dev *dev, uint32_t rate,
 /**
  * Set channel rate.
  * @param dev - The device structure.
- * @param chan - Channel number.
+ * @param chan_num - Channel number.
  * @param rate - Channel rate.
  * @return SUCCESS in case of success, negative error code otherwise.
  */
-int32_t hmc7044_clk_set_rate(struct hmc7044_dev *dev, uint32_t chan,
+int32_t hmc7044_clk_set_rate(struct hmc7044_dev *dev, uint32_t chan_num,
 			     uint64_t rate)
 {
 	uint32_t div;
 	int32_t ret;
+	struct hmc7044_chan_spec *chan = NULL;
 
-	if (chan >= dev->num_channels)
+	/* Find the reqested channel number */
+	for (int i=0; i < dev->num_channels; i++) {
```

Comment:
Declare variables at the beginning of the function

Reply:
Fixed